File: MasterApp/app/src/main/java/br/com/ysimplicity/masterapp/Helper/JsBridge/JsBridge.java
package br.com.ysimplicity.masterapp.helper.jsbridge;

import android.content.Context;
import android.support.annotation.NonNull;
import android.webkit.JavascriptInterface;

import br.com.ysimplicity.masterapp.helper.SessionManager;
import br.com.ysimplicity.masterapp.helper.jsbridge.JsListener;

/**
 * Created by Indigo on 11/18/16.
 */

public class JsBridge {

    private Context context;
    private JsListener listener;


    public JsBridge(@NonNull Context context, @NonNull JsListener listener) {
        this.context = context;
        this.listener = listener;
    }

    @JavascriptInterface
    public void postMessage(String message) {
        SessionManager.saveToken(context, message);
        listener.loginSuccessful();
    }

}

File: MasterApp/app/src/main/java/br/com/ysimplicity/masterapp/helper/jsbridge/JsBridge.kt
package br.com.ysimplicity.masterapp.helper.jsbridge

import android.content.Context
import android.webkit.JavascriptInterface
import br.com.ysimplicity.masterapp.helper.SessionManager

/**
 * Created by Ghost on 14/11/2017.
 */
class JsBridge(val context: Context, val listener: JsListener) {
    @JavascriptInterface
    fun postMessage(message: String) {
        SessionManager.saveToken(context, message)
        listener.loginSuccessful()
    }
}